spec/unit/gdal/string_list_spec.rb

```diff
@@ -0,0 +1,28 @@
+# frozen_string_literal: true
+
+require "ffi-gdal"
+require "gdal"
+
+RSpec.describe GDAL::StringList do
+  context "when no strings are provided" do
+    it "returns a new instance of StringList" do
+      subject { described_class.new }
+
+      expect(subject).to be_a(described_class)
+      expect(subject.c_pointer).to be_a(FFI::AutoPointer)
+      expect(subject.c_pointer.null?).to be(true)
```

Comment:
I like using predicate matchers in cases like this
```suggestion
      expect(subject.c_pointer).to be_null
```